src/azul/chalice.py

```diff
@@ -398,33 +400,36 @@ def __authenticate(self):
             log.info('Authenticated request as %r', auth)
 
     def _log_request(self):
-        if log.isEnabledFor(logging.INFO):
-            context = self.current_request.context
-            request_info = {
-                'query': self.current_request.query_params,
-                'headers': self.current_request.headers
-            }
-            log.info('Received %s request for %r, with %s.',
-                     context['httpMethod'],
-                     context['path'],
-                     json.dumps(request_info, cls=self._LogJSONEncoder))
+        context = self.current_request.context
+        request_info = {
+            'query': self.current_request.query_params,
+            'headers': self.current_request.headers
+        }
+        log.info('Received %s request for %r, with %s.',
+                 context['httpMethod'],
+                 context['path'],
+                 json.dumps(request_info, cls=self._LogJSONEncoder))
+        self._log_body_msg('request', self.current_request.json_body)
 
     def _log_response(self, response):
-        if log.isEnabledFor(logging.DEBUG):
-            n = 1024
-            if isinstance(response.body, str):
-                body = response.body[:n]
+        log.info('Returning %i response with headers %s.',
+                 response.status_code, json.dumps(response.headers, cls=self._LogJSONEncoder))
+        self._log_body_msg('response', response.body)
+
+    def _log_body_msg(self, body_type: str, body: str | JSON | None):
+        if body is not None:
+            body = json.dumps(body) if not isinstance(body, str) else body
+            if log.isEnabledFor(logging.DEBUG):
+                n = len(body)
+                len_msg = f' (total {n} bytes)'
+                if config.debug == 1:
+                    body = body[:1024]
+                    len_msg = f'{len_msg}, displaying initial 1024 bytes'
+                log.debug(http_body_log_message(body_type, body, verbatim=True) + len_msg + '.')
             else:
-                body = json_head(n, response.body)
-            log.debug('Returning %i response with headers %s. '
-                      'See next line for the first %i characters of the body.\n%s',
-                      response.status_code,
-                      json.dumps(response.headers, cls=self._LogJSONEncoder),
-                      n, body)
+                log.info('With %s body size of %i bytes.', body_type, len(body))
```

Comment:
```suggestion
                log.info('… with %s body size of %i bytes.', body_type, len(body))
```

Consider modifying `http_body_log_message()` so that it can return this message, perhaps by adding a `include_body: bool = True` parameter. This would keep all the different `… with` message strings in the same place.